youtube_dl/extractor/vxxx.py

```diff
@@ -0,0 +1,110 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+import base64
+import re
+
+from .common import InfoExtractor
+from ..utils import unified_timestamp, parse_duration
+
+
+class VXXXIE(InfoExtractor):
+    _VALID_URL = r'https?://vxxx\.com/video-(?P<id>\d+)'
+    _TESTS = [{
+        'url': 'https://vxxx.com/video-80747/',
+        'md5': '2f4bfd829b682ff9e3da1bda71b81b81',
+        'info_dict': {
+            'id': '80747',
+            'ext': 'mp4',
+            'title': 'Monica Aka Selina',
+            'display_id': 'monica-aka-selina',
+            'thumbnail': 'https://tn.vxxx.com/contents/videos_screenshots/80000/80747/420x236/1.jpg',
+            'description': '',
+            'timestamp': 1607167706,
+            'upload_date': '20201205',
+            'duration': 2373.0,
+            'categories': ['Anal', 'Asian', 'BDSM', 'Brunette', 'Toys',
+                           'Fetish', 'HD', 'Interracial', 'MILF'],
+            'age_limit': 18,
+        }
+    }]
+
+    def _download_info_object(self, video_id):
+        return self._download_json(
+            'https://vxxx.com/api/json/video/86400/0/{}/{}.json'.format(
+                int(video_id) // 1000 * 1000,
+                video_id,
+            ), video_id, headers={'Referer': 'https://vxxx.com'})['video']
+
+    def _download_format_object(self, video_id):
+        return self._download_json(
+            'https://vxxx.com/api/videofile.php?video_id={}'.format(video_id),
+            video_id,
+            headers={'Referer': 'https://vxxx.com'}
+        )
+
+    def _get_video_host(self):
+        return 'vxxx.com'
+
+    def _decode_base164(self, e):
+        """
+        Some non-standard encoding called "base164" in the JavaScript code. It's
+        similar to the regular base64 with a slightly different alphabet:
+            - "АВСЕМ" are Cyrillic letters instead of uppercase English letters
+            - "." is used instead of "+"; "," is used instead of "/"
+            - "~" is used for padding instead of "="
+        """
+
+        return base64.b64decode(e
+                                .replace("А", "A")
+                                .replace("В", "B")
+                                .replace("С", "C")
+                                .replace("Е", "E")
+                                .replace("М", "M")
+                                .replace(".", "+")
+                                .replace(",", "/")
+                                .replace("~", "=")
```

Comment:
Please make the "" '' (not required for doc string).

Would it be better to use the Unicode escapes ('\u0410\u0411\u0421\u0415\u041c') for the replacement ABCEM?

Or import `unicodedata` and use (eg) `unicodedata.lookup('CYRILLIC CAPITAL LETTER A')`

Also `import string` and:
```suggestion
        """
        Some non-standard encoding called "base164" in the JavaScript code. It's
        similar to the regular base64 with a slightly different alphabet:
            - "АВСЕМ" are Cyrillic letters instead of uppercase Latin letters
            - "." is used instead of "+"; "," is used instead of "/"
            - "~" is used for padding instead of "="
        """

            # using the kwarg to memoise the result
            def get_trans_tbl(from_, to, tbl={}):
                k = (from_, to)
                if not tbl.get(k):
                    tbl[k] = string.maketrans(from_, to)
                return tbl[k]

           # maybe for the 2nd arg:
           # import unicodedata and
           # ''.join((unicodedata.lookup('CYRILLIC CAPITAL LETTER ' + x) for x in ('A', 'BE', 'ES', 'IE', 'EM'))) + '+/='
           trans_tbl = get_trans_tbl('АBCEM.,~', 'ABCEM+/=')
           return base64.b64decode(e.translate(trans_tbl)
```